### DISCUSSION TITLE
How to export state to JSON

### DISCUSSION THREAD

**Discussion by sultaniman** [bcakmakoglu/vue-flow#1096] [Q&A] [answered]
Hey,

Thanks for your work! I am wondering on how can I export and import state from json?

**Comment by bcakmakoglu** [comment, marked as answer]
There are two methods you can use for this purpose
1. `toObject` - returns an object with the necessary info to save and restore a graph (i.e. nodes, edges and viewport position + zoom)
2. `fromObject` - takes an object that is returned by `toObject` and restores the graph to the state was was given

Both of these methods are available from `useVueFlow`
